### DISCUSSION TITLE
Next.js apollo server 4 and client subscriptions

### DISCUSSION THREAD

**Discussion by matthew-gordon** [enisdenjo/graphql-ws#428] [Q&A]
With the release of Apollo server 4, I would really like to use the as-integrations/next api route package along with graphql-ws to get a full stack chat app going in next.js 13 with subscriptions. I can’t think about how to actually accomplish this without building a custom server 🤔 would love to get a discussion going around this for examples

**Comment by enisdenjo** [comment]
I am not very familiar with Apollo Server 4 + NextJS 13. The [Enabling Subscriptions section in Apollo's documentation](https://www.apollographql.com/docs/apollo-server/data/subscriptions/#enabling-subscriptions) does not help?

**Comment by matthew-gordon** [comment]
It helps for stand alone servers, nextjs has a new paradigm api routes which allows for path based cloud functions (essentially serverless) while Apollo server 4 shifted to community maintained integrations. I have successful express/Koa integrations going with graphql-ws, specifically in this case using https://github.com/apollo-server-integrations/apollo-server-integration-next

I am just struggling with an SSR implementation or if it’s even possible 🤔

**Reply by enisdenjo** [reply]
If you could share a repo that has everything set up except for graphql-ws, I'd be more than happy to try integrating it.

Beware that during the server-side rendering in SSR you should not set up the client - only once the rendering starts happening in the browser.
